pkg/controller/cache/managed.go

```diff
@@ -199,6 +197,11 @@ func (e *external) Update(ctx context.Context, mg resource.Managed) (resource.Ex
 	if !ok {
 		return resource.ExternalUpdate{}, errors.New(errNotReplicationGroup)
 	}
+	// NOTE(muvaf): AWS API rejects modification requests if the state is not
+	// `available`
+	if cr.Status.AtProvider.Status != v1beta1.StatusAvailable {
+		return resource.ExternalUpdate{}, nil
+	}
```

Comment:
We could also handle this case by setting `UpToDate = true` in `Observe` and never calling update at all, but I like this pattern better because we're not actually up to date, but rather can't update yet.

Reply:
> but I like this pattern better because we're not actually up to date, but rather can't update yet.

Me too. Initially wanted the condition to be `status == modifying` but AWS explicitly says it has to be `available`. But I'm wondering if we could at least fire an event or note it down to somewhere that _we're not able to update the resource yet, here are some cat videos you can watch while waiting_

Reply:
> at least fire an event or note

We could potentially use a bespoke condition for this, either a variant of `Ready` or an entirely new condition type.